aip/client-libraries/4235.md

```diff
@@ -38,6 +38,13 @@ If the aforementioned requirements are met for a given field, client library
 generators **must** enable automatic population of said field in the
 generated client.
 
+The field **must not** be automatically populated by the client if:
+
+- The field does not support explicit presence, and its value is not equal to
+  the default
+- The field supports explicit presence, and presence indicates that it has been
+  set (even to the default value)
+
```

Comment:
Could I suggest we simplify the wording to avoid so many negatives? How about adding the following as one of the necessary conditions instead?:

```
The field must EITHER
- support explicit presence AND not have a value set by the user
OR
- not support explicit presence AND have its value be the default (empty string)
```

Reply:
I'm fine with changing it in that direction, but I'd reword it as:

> The field **must** be autopopulated if and only if one of the following conditions holds:
> - The field supports explicit presence and has *not* been set by the user
> - The field doesn't support explicit presence, and its value is the empty string (i.e. the default value)

Reply:
Thanks for the suggestion! That helps clarify things. I've updated it-